src/tests/unit/mock-helpers/mock-module-helpers.tsx

```diff
@@ -9,7 +9,7 @@ export const expectMockedComponentPropsToMatchSnapshots = (
     components.forEach(component => {
         const componentSnapshotName =
             snapshotName ||
-            `${component.displayName || component.name || 'mocked component'} props`;
+            `${component.displayName || component.name || component?.render?.displayName || 'mocked component'} props`;
```

Comment:
We need to calculate the component name the same way we do in `mockReactComponent` below in order to have it respect the passed in name.

Reply:
Updated